src/cmd/ksh93/bltins/test.c

```diff
@@ -545,7 +545,21 @@ int test_binop(Shell_t *shp, int op, const char *left, const char *right) {
         case TEST_LE: {
             return lnum <= rnum;
         }
-        default: { abort(); }
+        case TEST_REP: {
+            // The ksh93u release treated this as a silent failure. That is, `test foo =~ foo` would
+            // simply return zero (false) regardless of the operands and whether the condition is
+            // true. We now alert the user that the `=~` binop is not supported by the POSIX test
+            // command. See https://github.com/att/ast/issues/1152.
+            errormsg(SH_DICT, ERROR_exit(2), e_badop, "=~");
```

Comment:
May be we can provide a more helpful error message here. It should suggest to use `[[` operator.

Reply:
Good idea. Done.